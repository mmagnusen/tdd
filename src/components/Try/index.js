import React, { Component } from 'react';
import API from '../API'

class Try extends Component {
    state = {
        randomNumber: 100,
        multiplier: null
    }

    calculate = () => {
        try {
           this.setState({
            multiplier: 5
           })
        }
        finally {
            console.log('finally statement');
        }
    }



  render() {
    const { randomNumber, multiplier } = this.state;
    const { onClick } = this.props
    return (
    <div className="Intercept">
        <h1 className='fudge'>Random Number: {randomNumber}</h1>
        {multiplier && <h1 className='fudge'>Multiplier: {multiplier}</h1>}
        {multiplier && <h1 className='fudge'>Result: {randomNumber * multiplier}</h1>}
        <button onClick={this.calculate} id='calculate'>Try me</button>
        
    </div>
  );
}}

export default Try;
