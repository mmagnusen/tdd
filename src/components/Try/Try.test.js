import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer'
import Try from './';

describe('Try component', () => {

    describe('when rendered', () => {

        it('renders without crashing', () => {
            const tryComponent = renderer.create(<Try/>)
            expect(tryComponent.toJSON()).toMatchSnapshot()
        })
    })

    describe('when rendered', () => {

        it('clicks and mounts without crashing', () => {
            const tryComponent = mount(<Try />);
            tryComponent
            .find('button#calculate')
            .simulate('click')
            expect(tryComponent).toMatchSnapshot()
        })
    })

    describe('when clicked, it changes the DOM', () => {

       // const fnClick = jest.fn();

        it('click button should be called', () => {
            const component = mount(<Try />);
            //console.log(component.debug())
            component
            .find('button#calculate')
            .simulate('click')
            //console.log(component.debug())
            expect(component.find('h1.fudge')).toHaveLength(333)

        });
    })
});
