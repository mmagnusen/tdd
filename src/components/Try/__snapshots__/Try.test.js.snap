// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Try component when rendered clicks and mounts without crashing 1`] = `ReactWrapper {}`;

exports[`Try component when rendered renders without crashing 1`] = `
<div
  className="Intercept"
>
  <h1
    className="fudge"
  >
    Random Number: 
    100
  </h1>
  <button
    id="calculate"
    onClick={[Function]}
  >
    Try me
  </button>
</div>
`;
